### PR TITLE
Mayyar Karkout 204641

### PR DESCRIPTION
**Pull request by MeringueMayar**

### CODE REVIEW

**Comment by MeringueMayar** [comment]
3 minutes before midnight :)

File: src/test/java/edu/kis/vh/nursery/RyhmersJUnitTest.java [hunk 1]
```diff
@@ -76,3 +76,4 @@ public void testCountOut() {
     }
 
 }
+// alt+right/left arrow - Expand selection to next / previous element
```
Comment (kowallus) [inline src/test/java/edu/kis/vh/nursery/RyhmersJUnitTest.java]: It is not an Eclipse behaviour (lack of info about IDE used)

File: src/main/java/edu/kis/vh/nursery/DefaultCountingOutRyhmer.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
+package edu.kis.vh.nursery;
+
+public class DefaultCountingOutRyhmer {
+    private final int CAPACITY = 12;
+    private final int EMPTY_STACK_VALUE = -1;    
+    private int[] NUMBERS = new int[CAPACITY];
```
Comment (kowallus) [inline src/main/java/edu/kis/vh/nursery/DefaultCountingOutRyhmer.java:6]: numbers is not a real constant (lowercase)

File: src/main/java/edu/kis/vh/nursery/DefaultCountingOutRyhmer.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            return getEMPTY_STACK_VALUE();
+        }
+        return getNUMBERS()[total];
+    }
+
+    public int countOut() {
+        if (callCheck()) {
+            return getEMPTY_STACK_VALUE();
+        }
+        return getNUMBERS()[total--];
```
Comment (kowallus) [inline src/main/java/edu/kis/vh/nursery/DefaultCountingOutRyhmer.java:34]: getters should not be generated for internal (private) fields.

File: src/main/java/edu/kis/vh/nursery/DefaultCountingOutRyhmer.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    public int getTotal() {
+        return total;
+    }
+
+    /**
+     * @return the CAPACITY
+     */
+    public int getCAPACITY() {
+        return CAPACITY;
```
Comment (kowallus) [inline src/main/java/edu/kis/vh/nursery/DefaultCountingOutRyhmer.java:45]: Constants usually do not have getters.

File: src/main/java/edu/kis/vh/nursery/DefaultCountingOutRyhmer.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            getNUMBERS()[++total] = in;
+        }
+    }
+
+    public boolean callCheck() {
+        return total == getEMPTY_STACK_VALUE();
+    }
+
+    public boolean isFull() {
+        return total == 11;
```
Comment (kowallus) [inline src/main/java/edu/kis/vh/nursery/DefaultCountingOutRyhmer.java:20]: Another constant. It is related to CAPACITY

File: src/main/java/edu/kis/vh/nursery/FIFORyhmer.java [hunk 6]
```diff
@@ -1,21 +1,29 @@
...
-		
-		int ret = temp.countOut();
-		
-		while (!temp.callCheck())
-			
-		countIn(temp.countOut());
-		
-		return ret;
-	}
+    private DefaultCountingOutRyhmer defaultCountingOutRyhmer = new DefaultCountingOutRyhmer();
```
Comment (kowallus) [inline src/main/java/edu/kis/vh/nursery/FIFORyhmer.java:9]: can be final

File: src/main/java/edu/kis/vh/nursery/list/Node.java [hunk 7]
```diff
@@ -2,11 +2,15 @@
 
 public class Node {
```
Comment (kowallus) [inline src/main/java/edu/kis/vh/nursery/list/Node.java:3]: Node class should be refactored in 3.1.12

File: src/main/java/edu/kis/vh/nursery/list/IntLinkedList.java [hunk 8]
```diff
@@ -2,39 +2,41 @@
...
-
-	public int pop() {
-		if (isEmpty())
-			return -1;
-		int ret = last.value;
-		last = last.prev;
-		return ret;
-	}
+    private Node last;
+    private final int EMPTY = -1;
```
Comment (kowallus) [inline src/main/java/edu/kis/vh/nursery/list/IntLinkedList.java:6]: and static!

**Review by kowallus — COMMENTED** [review]
4 pts

**Pull request closed without merging**